app/controllers/csv_preview_controller.rb

```diff
@@ -7,10 +7,9 @@ class CsvPreviewController < ApplicationController
   rescue_from GdsApi::HTTPForbidden, with: :access_limited
 
   def show
-    @asset = GdsApi.asset_manager.whitehall_asset(legacy_url_path).to_hash
+    @asset = params[:legacy] ? GdsApi.asset_manager.whitehall_asset(path).to_hash : GdsApi.asset_manager.asset(params[:id]).to_hash
```

Comment:
What do you think about moving this out to a `before` action? Called something like `fetch_asset`  which runs before the `show` action only, and which sets @asset